File: server/router/business/books.go
package business

import (
	"github.com/giles-wong/zhongx-gva/server/middleware"
	"github.com/gin-gonic/gin"
)

type BooksRouter struct {
}

func (b *BooksRouter) InitBooksRouter(Router *gin.RouterGroup) {
	booksRouter := Router.Group("books").Use(middleware.OperationRecord())
	booksRouterWithoutRecord := Router.Group("books")

	{
		booksRouter.POST("addBook", booksApi.CreateBooks) // 新建Books
		booksRouter.POST("editBook", booksApi.EditBooks)
		booksRouter.DELETE("deleteBook", booksApi.DeleteBooks) // 删除Books
		//booksRouter.DELETE("deleteBooksByIds", booksApi.DeleteBooksByIds) // 批量删除Books
	}
	{
		booksRouterWithoutRecord.POST("getBookList", booksApi.BookList)
	}
}

File: web/src/api/books.js
import service from '@/utils/request'

export const getBooksList = (data) => {
    return service({
        url: '/books/getBookList',
        method: 'post',
        data: data
    })
}

export const createBook = (data) => {
    return service({
        url: '/books/addBook',
        method: 'post',
        data: data
    })
}

export const editBook = (data) => {
    return service({
        url: '/books/editBook',
        method: 'post',
        data: data
    })
}

export const deleteBook = (data) => {
    return service({
        url: '/books/deleteBook',
        method: 'delete',
        data: data
    })
}